Finalize lts masks

mkutu opened the pull request:
Setup logic to save all "finalized" or "good" masks to LTS whether they are the `initial`, `refined`, or `relabeled` mask, and update the db with all the necessary fields.

- [x] Mark any masks with "good" tags as "finalized"  mask_status in the temporary db. Do this during `inspect`.
- [x] Create logic in `finalize` mode so that all "finalized" masks are sent to LTS
- [x] Cutouts got to ` /mnt/research-projects/r/raatwell/longterm_images3/field-cutouts/`
- [x] Final masks go to ` /mnt/research-projects/r/raatwell/longterm_images3/field-masks/`
- [x] Consolidate refined and relabeled products into a respective folders
- [x] store refined and relabeled cutout, cutout mask, and fullsize mask in temp db
  -  this allow us to refer to it later when we move it to LTS

Note: Use `/mnt/research-projects/r/raatwell/longterm_images3/field-tools/agir_field_db/agir_field_test.db` for testing.

mkutu committed:
Enhance mask finalization logic and add file existence checks

mkutu committed:
Enhance mask finalization logic and add file existence checks

navjot-nangia commented:
Notes for self:
- 'mask_status' column already get set to 'finalized' if inspection tag is given 'good'
- for relabeled images, the 'mask_status' column has 'relabeled' tag. It should be changed to 'finalized'
- there is another column with name 'final_mask_issue_tag'. It has 'good' tags from inspection results. But no tags from cvat relabeling. This column name is deceiving.
- SHOULD 'mask_status' OR 'final_mask_issue_tag' BE USED TO SEND FINAL MASKS TO LTS?

Going to make changes to the 'final_mask_issue_tag'. Will put 'finalized' for 'good' initial tags (in task INSPECT) and 'relabeled' mask_status (in task EXPORT_CVAT)

navjot-nangia committed:
- added 'unreviewed' to ignore in mask_status
- made sure the "initial_tag" and "reviewer" from fiftyone samples is handled correctly

navjot-nangia committed:
minor

navjot-nangia committed:
minor: changed relabelled to relabeled for consistency

mkutu committed:
Merge branch 'finalize-lts-masks' of https://github.com/precision-sustainable-ag/Field-SegmentationTraining into finalize-lts-masks

navjot-nangia committed:
minor: for consistency

navjot-nangia committed:
Ensure string conversion for mask issue tags and metadata in CVAT sample creation

navjot-nangia committed:
marking 'finalized' for 'final_mask_tag' if inspection tag is 'good'

navjot-nangia committed:
- Fixed typo in STATUS_RELABELLED to STATUS_RELABELED
- made sure nan values in final_tag are taken care of

navjot-nangia committed:
minor

navjot-nangia committed:
Fix typo in mask status from 'relabelled' to 'relabeled'

navjot-nangia committed:
Fix typo in mask status from 'relabelled' to 'relabeled'

mkutu committed:
Refactor mask generation paths and update related methods for clarity and consistency

mkutu committed:
Merge branch 'db2csv' into finalize-lts-masks
